rustls/tests/api.rs

```diff
@@ -5736,56 +5683,112 @@ fn test_client_removes_tls12_session_if_server_sends_undecryptable_first_message
     ));
 }
 
-#[cfg(all(feature = "ring", not(feature = "fips")))]
 #[test]
 fn test_client_fips_service_indicator() {
-    assert!(!make_client_config(KeyType::Rsa).fips());
+    if provider_is_fips() {
+        assert!(make_client_config(KeyType::Rsa).fips());
+    } else {
+        assert!(!make_client_config(KeyType::Rsa).fips());
+    }
 }
 
-#[cfg(all(feature = "ring", not(feature = "fips")))]
 #[test]
 fn test_server_fips_service_indicator() {
-    assert!(!make_server_config(KeyType::Rsa).fips());
+    if provider_is_fips() {
+        assert!(make_server_config(KeyType::Rsa).fips());
+    } else {
+        assert!(!make_server_config(KeyType::Rsa).fips());
+    }
 }
 
-#[cfg(feature = "fips")]
 #[test]
-fn test_client_fips_service_indicator() {
-    assert!(make_client_config(KeyType::Rsa).fips());
+fn test_client_fips_service_indicator_includes_require_ems() {
+    if provider_is_fips() {
+        let mut client_config = make_client_config(KeyType::Rsa);
+        assert!(client_config.fips());
+        client_config.require_ems = false;
+        assert!(!client_config.fips());
+    }
 }
 
-#[cfg(feature = "fips")]
 #[test]
-fn test_server_fips_service_indicator() {
-    assert!(make_server_config(KeyType::Rsa).fips());
+fn test_server_fips_service_indicator_includes_require_ems() {
+    if provider_is_fips() {
+        let mut server_config = make_server_config(KeyType::Rsa);
+        assert!(server_config.fips());
+        server_config.require_ems = false;
+        assert!(!server_config.fips());
+    }
 }
 
-#[cfg(feature = "fips")]
-#[test]
-fn test_client_fips_service_indicator_includes_require_ems() {
-    let mut client_config = make_client_config(KeyType::Rsa);
-    assert!(client_config.fips());
-    client_config.require_ems = false;
-    assert!(!client_config.fips());
+} // test_for_each_provider!
+
+use std::cell::RefCell;
+
+thread_local!(static COUNTS: RefCell<LogCounts> = RefCell::new(LogCounts::new()));
+
+struct CountingLogger;
```

Comment:
Feels like this could/should be a separate commit.

Reply:
have moved this to the bottom of the file in a prior commit